src/server/core/dcitem.cpp

```diff
@@ -1042,8 +1042,16 @@ bool DCItem::transform(ItemValue &value, time_t nElapsedTime)
                value = value.getUInt32() - m_prevRawValue.getUInt32();
                break;
             case DCI_DT_COUNTER32:
-               // assume counter reset if new value is less then previous
-               value = (value.getUInt32() > m_prevRawValue.getUInt32()) ? value.getUInt32() - m_prevRawValue.getUInt32() : 0;
+	       if (value.getUInt32() >= m_prevRawValue.getUInt32())
+	       {
+		  value = value.getUInt32() - m_prevRawValue.getUInt32();
+	       }
+	       else
+	       {
+		  // discontinuity: counter reset or wraparound, return error per NX-2461
+		  success = false;
```

Comment:
This won't work if transformation script is present (transformation block will reset success back to true)